scripts/sign_encrypt.py

```diff
@@ -29,391 +59,876 @@ def int_parse(str):
+            e_bytes = find_attr(TEE_ATTR_RSA_PUBLIC_EXPONENT)
+            e = int.from_bytes(e_bytes, 'big')
+            n = int.from_bytes(n_bytes, 'big')
+            self.subkey_key = rsa.RSAPublicNumbers(e, n).public_key()
+
+            self.img = self.inf[subkey_offs:offs - self.name_size]
+            if len(self.img) != img_size:
+                raise Exception("Unexpected img size: got {}, expected {}"
+                                .format(len(self.img), img_size))
+        else:
+            raise Exception("Unsupported image type: {}".format(img_type))
+
+
+    def display(self):
+        import binascii
+        import struct
+        import uuid
+
+        def display_ta():
+            nonlocal offs
+            ta_uuid = self.inf[offs:offs + UUID_SIZE]
+            print('  uuid:       {}'.format(uuid.UUID(bytes=ta_uuid)))
+            offs += UUID_SIZE
+            [ta_version] = struct.unpack('<I', self.inf[offs:offs + 4])
+            print('  ta_version: {}'.format(ta_version))
+
+            offs += 4;
+            if img_type == SHDR_ENCRYPTED_TA:
+                ehdr = self.inf[offs: offs + EHDR_SIZE]
+                offs += EHDR_SIZE
+                [enc_algo, flags, nonce_len,
+                 tag_len] = struct.unpack('<IIHH', ehdr)
+
+                enc_algo_name = 'Unkown'
+                if enc_algo in enc_tee_alg.values():
+                    enc_algo_name = value_to_key(enc_tee_algo, enc_algo)
+                print('  enc_algo:   {} ({})'.format(enc_algo, enc_algo_name))
+
+                if enc_algo not in enc_tee_alg.values():
+                    raise Exception('Unrecognized encrypt algorithm: 0x{:08x}'
+                                    .format(enc_value))
+                print('  nonce_len:  {} (bytes)'.format(nonce_len))
+                if nonce_len != NONCE_SIZE:
+                    raise Exception("Unexpected nonce len: {}"
+                                    .format(nonce_len))
+                nonce = self.inf[offs:offs + nonce_len]
+                print('  nonce:      {}...'.format(binascii.hexlify(nonce)))
+                offs += nonce_len
+
+                print('  tag_len:    {} bytes'.format(tag_len))
+                if tag_len != TAG_SIZE:
+                    raise Exception("Unexpected tag len: {}".format(tag_len))
+                tag = self.inf[-tag_len:]
+                print('  tag:        {}...'.format(binascii.hexlify(tag)))
+                ciphertext = self.inf[offs:-tag_len]
+                print('  Ciph offset:{} (0x{:x}) bytes)'.format(offs, offs))
+                print('  Ciph size:  {} (0x{:x}) (bytes)'
+                      .format(len(ciphertext), len(ciphertext)))
+                if len(ciphertext) != img_size:
+                    raise Exception("Unexpected ciphertext size: ",
+                                    "got {}, expected {}"
+                                    .format(len(ciphertext), img_size))
```

Comment:
Would it make sense not to make a special case of this and print "TA offset" and "TA size" like in the else part? It is ciphertext, but it is also the TA;  the fact it is encrypted doesn't matter much and it is shown by the `SHDR_ENCRYPTED_TA` type anyways.